### PR TITLE
fix calendar bug

### PR DESCRIPTION
**Pull request by KayMKM**
## Description
fix some bugs

## Related Issue
1. state won't clear when update meeting again
2. some ambiguous wordings in update and delete flow
3. only show the start date when show the cross-days meeting
4. when the meeting title contains '"', rendering card will go wrong.
5. won't show the meetings are past when ask "show meetings from 4pm to 9pm"

## Testing Steps
1. input "update my 5pm meeting to 6pm" -> "cancel" -> "update my meeting"
bot should ask for time again
2. input "update my 5pm meeting"(you don't have meeting at 5pm)
bot shouldn't say "It seems there is no event with that name, try again please."
input "update my test meeting"(you don't have meeting that's title is "test")
bot shouldn't say "It seems there is no event at this time, please try again."
3. create a cross-days meeting and ask bot to show it. the adaptive card will show the start date and end date
4. create a meeting with a title contains '"', then ask bot to show it.
5. input "show meetings from 4pm to 9pm"(current time is after 4pm), bot will show all meetings in 4-9pm.

### CODE REVIEW

File: solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/AdaptiveCards/AdaptiveCardHelper.cs [hunk 1]
```diff
@@ -42,8 +42,8 @@ public static AdaptiveCard GetCardFromJson(string jsonFile, StringDictionary tok
                 var escapedTokens = new StringDictionary();
                 foreach (string key in tokens.Keys)
                 {
-                    var escapedTokenStr = tokens[key]?.Replace("\"", "\\\"");
-                    escapedTokenStr = escapedTokenStr?.Replace("\\", "\\\\");
+                    var escapedTokenStr = tokens[key]?.Replace("\\", "\\\\");
+                    escapedTokenStr = escapedTokenStr?.Replace("\"", "\\\"");
```
Comment (lzc850612) [inline solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/AdaptiveCards/AdaptiveCardHelper.cs:47]: can you give us an example on how this change is fixing? seems like you changed the order of these two replacement operations but i don't know how that's going to help.

**Review by lzc850612 — CHANGES_REQUESTED** [review]

File: solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/AdaptiveCards/AdaptiveCardHelper.cs (hunk 1 above)
Comment (KayMKM) [reply solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/AdaptiveCards/AdaptiveCardHelper.cs:47]: @lzc850612 this is fixing the forth issue: when the meeting title contains '"', rendering card will go wrong. We expect "\\" convert to "\\\\" and "\"" convert to "\\\"". If we didn't change the order, the "\"" will change to "\\\"" first and then convert to "\\\\\"". So we should convert "\\" first.

Comment (lzc850612) [reply solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/AdaptiveCards/AdaptiveCardHelper.cs:47]: Ok. Can you please add some comments describing this issue you're saving in the code? this way it's easier for people to understand what you're trying to achieve

Comment (KayMKM) [reply solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/AdaptiveCards/AdaptiveCardHelper.cs:47]: @lzc850612 comment added

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Models/EventModel.cs [hunk 2]
```diff
@@ -648,9 +648,16 @@ public CalendarCardData ToAdaptiveCardData(TimeZoneInfo timeZone, bool showDate
...
+            if (showDate || !IsSameDate(eventItem.StartTime, eventItem.EndTime))
             {
-                textString += $"\n{TimeConverter.ConvertUtcToUserTime(eventItem.StartTime, timeZone).ToString("dd-MM-yyyy")}";
+                if (IsSameDate(eventItem.StartTime, eventItem.EndTime))
+                {
+                    textString += $"\n{TimeConverter.ConvertUtcToUserTime(eventItem.StartTime, timeZone).ToString("dd-MM-yyyy")}";
+                }
+                else
+                {
+                    textString += $"\n{TimeConverter.ConvertUtcToUserTime(eventItem.StartTime, timeZone).ToString("dd-MM-yyyy")} - {TimeConverter.ConvertUtcToUserTime(eventItem.EndTime, timeZone).ToString("dd-MM-yyyy")}";
```
Comment (lzc850612) [inline solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Models/EventModel.cs]: define

var userTimeString = $"\n{TimeConverter.ConvertUtcToUserTime(eventItem.StartTime, timeZone)";

and use it in both conditions

and why does the string start with \n ?

**Review by lzc850612 — CHANGES_REQUESTED** [review]

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Models/EventModel.cs (hunk 2 above)
Comment (KayMKM) [reply solutions/Virtual-Assistant/src/csharp/skills/calendarskill/Models/EventModel.cs]: @lzc850612 had changed in new commit. As designed, we need date in a new line of the card content. But in some case the date won't show. So I linked these strings dynamically here.

**Review by lzc850612 — APPROVED** [review]

**Pull request merged**